dataset/src/test/scala/frameless/functions/UnaryFunctionsTest.scala

```diff
@@ -0,0 +1,40 @@
+package frameless
+package functions
+
+import org.scalacheck.Prop
+import org.scalacheck.Prop._
+
+import scala.math.Ordering
+
+class UnaryFunctionsTest extends TypedDatasetSuite {
+  test("size on vector test") {
+    def prop[A: TypedEncoder](xs: List[X1[Vector[A]]]): Prop = {
+      val tds = TypedDataset.create(xs)
+
+      val framelessResults = tds.select(size(tds('a))).collect().run().toVector
+      val scalaResults = xs.map(x => x.a.size).toVector
+
+      framelessResults ?= scalaResults
+    }
+
+    check(forAll(prop[Long] _))
+    check(forAll(prop[Char] _))
+    check(forAll(prop[X2[Int,Option[Long]]] _))
+  }
+
+  test("sort on vector test") {
+    def prop[A: TypedEncoder: Ordering](xs: List[X1[Vector[A]]]): Prop = {
```

Comment:
I think there is also supposed to be a space before the second `:` (otherwise it loops like you have typing `TypedEncoder` and not `A` :P)